Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -330,13 +347,27 @@ protected virtual Vector3 CalculateGoalPosition()
             if (trackedController != null &&
                 handBounds.Bounds.TryGetValue(trackedController.ControllerHandedness, out trackedHandBounds))
             {
-                float distance;
-                Ray ray = CalculateProjectedSafeZoneRay(goalPosition, SolverHandler.TransformTarget, trackedController, safeZone, OffsetBehavior);
-                trackedHandBounds.Expand(safeZoneBuffer);
-
-                if (trackedHandBounds.IntersectRay(ray, out distance))
+                // If we are mounting to the palm, and don't want to use raycasting to find the
+                // palm goal position, we will calculate the goal position using the palm joint directly.
+                if (safeZone == SolverSafeZone.AtopPalm && !UsePalmRaycast)
+                {
+                    MixedRealityPose? palmPose = GetPalmPose(trackedController);
```

Comment:
Can you think of a case when somebody would want to use AtopPalm, but not UsePalmRaycast? If not, perhaps you can just get rid of UsePalmRaycast to make the code and configuration a little bit simpler. As it stands, UsePalmRaycast bool is only valid for a safeZone == AtopPalm.

Reply:
This is a bit tricky. The hand safe zone targeting system typically uses a raycast method to generate safe zone targets; however, the bounds used in the raycast method are not very precise, and are too unstable for something like precise palm positioning.

Therefore, the majority of the time, you'd want to disable this raycast system when using palm mounts.

Any other safe zone target can/should still use the raycast system.

Reply:
@julenka This has been amended in the new commit. As the raycast system has been overhauled/improved to use local computations, the raycast is now sufficiently accurate for palm positioning.